pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test: TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr failed

pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test.TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20616228?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20616228?buildTab=artifacts#/) on master @ [1c5a706b720356871c8a262c8adae1cef3cdfee3](https://github.com/cockroachdb/cockroach/commits/1c5a706b720356871c8a262c8adae1cef3cdfee3):


```
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8554/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 3m30.02107046s
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8554/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 3m40.021175442s
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8554/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 3m50.021767523s
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8554/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 4m0.021878658s
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8554/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 4m10.022031727s
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8554/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 4m20.022174465s
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8554/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 4m30.022412792s
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8554/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 4m40.024270258s
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8554/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 4m50.024429192s
    panic.go:635: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/9b0e3b994513aa3a64691c2213aad339/logTestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr1511531362
--- FAIL: TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr (293.34s)
=== RUN   TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_2_of_24
--- FAIL: TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_2_of_24 (202.87s)
=== RUN   TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_2_of_24/restore_all_tables_in_database
    sql_runner.go:117: 
        	Error Trace:	pkg/testutils/sqlutils/sql_runner.go:130
        	            				pkg/testutils/sqlutils/sql_runner.go:391
        	            				pkg/sql/schemachanger/sctest/framework.go:976
        	            				pkg/sql/schemachanger/sctest/backup.go:587
        	Error:      	Received unexpected error:
        	            	query 'SELECT status, job_type, description FROM [SHOW JOBS] WHERE job_type IN ('SCHEMA CHANGE', 'TYPEDESC SCHEMA CHANGE', 'NEW SCHEMA CHANGE') AND status NOT IN ('succeeded', 'failed')': expected:
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:394
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	pkg/testutils/soon.go:71
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	pkg/util/retry/retry.go:355
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	pkg/testutils/soon.go:77
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:130
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:391
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/sctest.waitForSchemaChangesToFinish
        	            	  | 	pkg/sql/schemachanger/sctest/framework.go:976
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/sctest.exerciseBackupRestore.func1
        	            	  | 	pkg/sql/schemachanger/sctest/backup.go:587
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1690
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_s390x.s:797
        	            	Wraps: (2) query 'SELECT status, job_type, description FROM [SHOW JOBS] WHERE job_type IN ('SCHEMA CHANGE', 'TYPEDESC SCHEMA CHANGE', 'NEW SCHEMA CHANGE') AND status NOT IN ('succeeded', 'failed')': expected:
        	            	  |
        	            	  | got:
        	            	  | running, NEW SCHEMA CHANGE, ALTER TABLE multiregion_db.public.table_regional_by_row ADD … NG HASH
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
--- FAIL: TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_2_of_24/restore_all_tables_in_database (55.94s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-55528

pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test.TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20621749?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20621749?buildTab=artifacts#/) on master @ [cf01ffa5e9ecc35b63d7228cf113b74bcb809751](https://github.com/cockroachdb/cockroach/commits/cf01ffa5e9ecc35b63d7228cf113b74bcb809751):


```
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7484/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 50.008916491s
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7484/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 1m0.010276051s
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7484/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 1m10.01073474s
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7484/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 1m20.011418364s
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7484/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 1m30.011727088s
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7484/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 1m40.012089601s
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7484/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 1m50.013143493s
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7484/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 2m0.013458226s
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7484/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 2m10.013603387s
    panic.go:635: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/9b0e3b994513aa3a64691c2213aad339/logTestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr661457593
--- FAIL: TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr (137.01s)
=== RUN   TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_1_of_24/restore_all_tables_in_database
    sql_runner.go:117: 
        	Error Trace:	pkg/testutils/sqlutils/sql_runner.go:130
        	            				pkg/testutils/sqlutils/sql_runner.go:391
        	            				pkg/sql/schemachanger/sctest/framework.go:976
        	            				pkg/sql/schemachanger/sctest/backup.go:587
        	Error:      	Received unexpected error:
        	            	query 'SELECT status, job_type, description FROM [SHOW JOBS] WHERE job_type IN ('SCHEMA CHANGE', 'TYPEDESC SCHEMA CHANGE', 'NEW SCHEMA CHANGE') AND status NOT IN ('succeeded', 'failed')': expected:
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:394
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	pkg/testutils/soon.go:71
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	pkg/util/retry/retry.go:355
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	pkg/testutils/soon.go:77
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:130
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:391
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/sctest.waitForSchemaChangesToFinish
        	            	  | 	pkg/sql/schemachanger/sctest/framework.go:976
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/sctest.exerciseBackupRestore.func1
        	            	  | 	pkg/sql/schemachanger/sctest/backup.go:587
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1690
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_s390x.s:797
        	            	Wraps: (2) query 'SELECT status, job_type, description FROM [SHOW JOBS] WHERE job_type IN ('SCHEMA CHANGE', 'TYPEDESC SCHEMA CHANGE', 'NEW SCHEMA CHANGE') AND status NOT IN ('succeeded', 'failed')': expected:
        	            	  |
        	            	  | got:
        	            	  | running, NEW SCHEMA CHANGE, ALTER TABLE multiregion_db.public.table_regional_by_row ADD … NG HASH
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
--- FAIL: TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_1_of_24/restore_all_tables_in_database (46.84s)
=== RUN   TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_1_of_24
--- FAIL: TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_1_of_24 (80.84s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/schemachanger/sctest/framework.go
+++ b/pkg/sql/schemachanger/sctest/framework.go
@@ -965,14 +965,13 @@ func waitForSchemaChangesToSucceed(t *testing.T, tdb *sqlutils.SQLRunner) {
 }
 
 func waitForSchemaChangesToFinish(t *testing.T, tdb *sqlutils.SQLRunner) {
-	// Wait longer on stress for schema changes.
-	if skip.Stress() {
-		old := tdb.SucceedsSoonDuration
-		tdb.SucceedsSoonDuration = time.Minute * 2
-		defer func() {
-			tdb.SucceedsSoonDuration = old
-		}()
-	}
+	// Schema changes in more complex tests can be slower, so give them
+	// a lot more headroom to complete.
+	old := tdb.SucceedsSoonDuration
+	tdb.SucceedsSoonDuration = time.Minute * 2
+	defer func() {
+		tdb.SucceedsSoonDuration = old
+	}()
 	tdb.CheckQueryResultsRetry(
 		t, schemaChangeWaitQuery(`('succeeded', 'failed')`), [][]string{},
 	)

